src/helpful_role/index.ts

```diff
@@ -1,16 +1,47 @@
-import { MessageReaction } from 'discord.js';
+import { MessageReaction, User } from 'discord.js';
 import { Document } from 'mongoose';
 
+import { IS_PROD } from '../env';
 import pointHandler from './point_handler';
 
-export interface User extends Document {
+/**
+ * If you are not sure what the unicode for a certain emoji is,
+ * consult the emojipedia. https://emojipedia.org/
+ */
+export const allowedEmojis = ['✅', '✔️', '☑️', '🆙', '⬆️', '⏫', '🔼'];
+
+export interface IUser extends Document {
   user?: string;
   points?: number;
 }
 
-const handleHelpfulRole = async (reaction: MessageReaction) => {
+const handleHelpfulRole = async (reaction: MessageReaction, user: User) => {
   // Break if the author of the message is trying to upvote his own solution
-  if (reaction.users.cache.find(u => u.id === reaction.message.author.id)) {
+  if (
+    IS_PROD && // Enable upvoting own messages for development purposes
+    reaction.users.cache.find(u => u.id === reaction.message.author.id)
+  ) {
+    return;
+  }
+
+  // Create a list of user IDs to conduct a check if the user has already upvoted the message with another emoji.
+  let users: string[] = [];
+
+  reaction.message.reactions.cache.forEach(r => {
+    if (
+      allowedEmojis
+        .filter(e => e !== reaction.emoji.name)
+        .includes(r.emoji.name)
+    ) {
+      users = [...users, ...r.users.cache.keyArray()];
+    }
+  });
```

Comment:
what about

```ts
const users = reaction.message.reactions.cache.map(reaction => {
    if (
      allowedEmojis
        .filter(e => e !== reaction.emoji.name)
        .includes(r.emoji.name)
    ) {
      return reaction.users.cache.keyArray();
    }

    return null;
}).filter(Boolean);
```
? :)

Reply:
This would end up returning a bunch of arrays, within an array. Each reaction would end up returning an array of IDs, and I am not sure if flattening the end `users` array is a smart idea.